src/SelectMenu/SelectMenu.js

```diff
@@ -67,7 +67,7 @@ const SelectMenu = React.forwardRef(({children, initialTab, as, ...rest}, forwar
   // handles the overlay behavior - closing the menu when clicking outside of it
   useEffect(() => {
     if (open) {
-      document.addEventListener('click', onClickOutside)
+      document.addEventListener('click', onClickOutside, {capture: true})
```

Comment:
See: https://reactjs.org/blog/2020/08/10/react-v17-rc.html#fixing-potential-issues

Reply:
In the case of the SelectMenu and Details element we want to know about _any_ click in the document so we need to add `capture: true`